appserver/deployment/dol/src/main/java/com/sun/enterprise/deployment/ApplicationClientDescriptor.java

```diff
@@ -165,25 +160,25 @@ public Collection getNamedDescriptors() {
     /**
     * Returns the set of environment properties of this app client.
     */
-    public Set getEnvironmentProperties() {
+    @Override
+    public Set<EnvironmentProperty> getEnvironmentProperties() {
 	if (this.environmentProperties == null) {
-	    this.environmentProperties = new OrderedSet();
+	    this.environmentProperties = new OrderedSet<>();
 	}
-	return this.environmentProperties = new OrderedSet(this.environmentProperties);
+	return this.environmentProperties = new OrderedSet<>(this.environmentProperties);
     }
 
     /**
      * Returns the environment property object searching on the supplied key.
      * throws an illegal argument exception if no such environment property exists.
      */
+    @Override
     public EnvironmentProperty getEnvironmentPropertyByName(String name) {
-	for (Iterator itr = this.getEnvironmentProperties().iterator();
-             itr.hasNext();) {
-	    EnvironmentProperty ev = (EnvironmentProperty) itr.next();
-	    if (ev.getName().equals(name)) {
-		return ev;
-	    }
-	}
+        for (EnvironmentProperty ev : this.getEnvironmentProperties()) {
```

Comment:
This is no longer thread safe, I'd revert this to be safe

Reply:
The collection is not being modified, so it doesn't need to create an iterator. All of the for-loop changes were suggested by the IDE.